internal/pkg/gateway/server/apiutils_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package server
+
+import (
+	"testing"
+
+	"github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetValueFromResponse(t *testing.T) {
+	response := createProposalResponse("MyResult", t)
+
+	result, err := getValueFromResponse(response)
+	require.NoError(t, err, "Failed to extract value from response")
+	require.Equal(t, []byte("MyResult"), result.Value, "Incorrect value")
+}
+
+func TestCreateUnsignedTx(t *testing.T) {
+	proposal := &peer.Proposal{
+		Header:  []byte{},
+		Payload: []byte{},
+	}
+	response := createProposalResponse("MyResult", t)
+	_, err := createUnsignedTx(proposal, response)
```

Comment:
Why aren't we verifying the result? The next line asserts no error implying we expect a good value. I'd assume value we're ignoring is the _raison d'étre_ of this function.

Put differently, asserting on no error is not the same as asserting we got back what we expect. There used to be lots of tests in Fabric that did this and we've slowly gone back to fix most of them.

edit: Looks like this applies to most of what's below?

Reply:
Actually, now that I've rewritten my implementation of createUnsignedTx to call down into protoutils, most of these tests are redundant (they duplicate test cases in txutils_test.go). So I'll cut this back to a single test.